src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -197,32 +229,56 @@ class CoreWorkerDirectTaskSubmitter {
   /// (1) The lease client through which the worker should be returned
   /// (2) The expiration time of a worker's lease.
   /// (3) The number of tasks that are currently in flight to the worker
-  /// (4) The resources assigned to the worker
-  /// (5) The SchedulingKey assigned to tasks that will be sent to the worker
+  /// (4) A boolean that indicates whether we have launched a StealTasks request, and we
+  /// are waiting for the stolen tasks (5) The resources assigned to the worker (6) The
+  /// SchedulingKey assigned to tasks that will be sent to the worker
   struct LeaseEntry {
     std::shared_ptr<WorkerLeaseInterface> lease_client;
     int64_t lease_expiration_time;
     uint32_t tasks_in_flight;
+    bool currently_stealing;
     google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry> assigned_resources;
     SchedulingKey scheduling_key;
 
     LeaseEntry(
         std::shared_ptr<WorkerLeaseInterface> lease_client = nullptr,
         int64_t lease_expiration_time = 0, uint32_t tasks_in_flight = 0,
+        bool currently_stealing = false, int64_t stolen_tasks_to_wait_for = 0,
```

Comment:
These values always seem to be the same on construction, so I would not expose them as args in the constructor. Can you instead initialize the struct members' values during declaration? Actually, same goes for the pre-existing. `tasks_in_flight` member.

Also it looks like `stolen_tasks_to_wait_for` is no longer used.

Reply:
yup!